src/wp-includes/style-engine/class-wp-style-engine-css-declarations.php

```diff
@@ -0,0 +1,191 @@
+<?php
+/**
+ * WP_Style_Engine_CSS_Declarations
+ *
+ * Holds, sanitizes and prints CSS rules declarations
+ *
+ * @package WordPress
+ * @subpackage StyleEngine
+ * @since 6.1.0
+ */
+
+/**
+ * Class WP_Style_Engine_CSS_Declarations.
+ *
+ * Holds, sanitizes, processes and prints CSS declarations for the style engine.
+ *
+ * @access private
+ * @since 6.1.0
+ */
+class WP_Style_Engine_CSS_Declarations {
```

Comment:
> or do we extend/plan to extend this class

I'm not sure of the strategy right now.

I expect that there is no plan to extend. Since this comes from a Gutenberg package, we expect to version it. 

Backwards compat will foil that I guess. 🤷 

Happy to take any advice onboard!

Reply:
@azaozz Do you have any thoughts on this?

Reply:
@costdev thanks for the ping.

If this is intended to be private, "core use only", it will perhaps have to be locked in a closure like the current Webfonts code is. That will ensure it can be updated in the future without concerns about backwards compatibility.

If not locked, back-compat will have to be maintained, even if the class is `final`.

Reply:
Thanks @azaozz! _If_ it's not locked and back-compat has to be maintained, I don't think there would be any benefit in having `@access private`. Would you agree?

Reply:
> it will perhaps have to be locked in a closure like the current Webfonts code is.

Thanks for the advice here.

I think the class in question (`WP_Style_Engine_CSS_Declarations`), and the following, other, new classes:

- `WP_Style_Engine_CSS_Rule`
- `WP_Style_Engine_CSS_Rules_Store`
- `WP_Style_Engine_Processor`

should be fairly stable. Maintaining backwards compatibility will be expected and, I hope, not overly tortuous. 

I'll remove `@access private` from these.

`WP_Style_Engine`, on the other hand, is the primary integration point and provides the bulk of the functionality.

I expect it will have to be updated more frequently, especially given that this is a new API.

In that case, do folks think it's safer to refactor `WP_Style_Engine` yo something similar to what's happening in webfonts? https://github.com/ramonjd/wordpress-develop/blob/bb0cbe3f557d16714ca109a28c2877d178adc94f/src/wp-includes/script-loader.php#L3059-L3058

cc @aristath and @andrewserong for thoughts